.github/workflows/build_and_test.yml

```diff
@@ -117,7 +117,15 @@ jobs:
             CARGO_FLAGS="--release --features profiling"
           fi
 
-          "${cov_prefix[@]}" mold -run cargo build $CARGO_FLAGS --features failpoints --bins --tests
+          "${cov_prefix[@]}" mold -run cargo build $CARGO_FLAGS --features failpoints --all --all-targets
+
+      - name: Check formatting
```

Comment:
One reason to put style checks into a separate workflow was the desire to run the e2e tests despite style errors.

Also, if we keep it this way, the `target/` directory will grow even bigger, since `cargo clippy` results are now stored there. I think they are already in that cache, by the way, since the Rust cache key is not changes and this PR's action was run.

WDYT about moving this into a separate step as it was initially in https://github.com/neondatabase/neon/pull/2079, before force pushes, where it was run along with e2e tests? Postgres build results could be restored from cache as it's done in `Get postgres artifact for restoration` here; Rust build results could be either cached separately or left without any caching, `cargo clippy` is not that slow to run compared to `cargo build`.